src/tmt_web/utils/git_handler.py

```diff
@@ -0,0 +1,116 @@
+import contextlib
+import os
+from shutil import rmtree
+
+from tmt import Logger
+from tmt.utils import (  # type: ignore[attr-defined]
+    Command,
+    Common,
+    GeneralError,
+    Path,
+    RunError,
+    git_clone,
+)
+
+
+def checkout_branch(path: Path, logger: Logger, ref: str) -> None:
+    """
+    Checks out the given branch in the repository.
+    :param ref: Name of the ref to check out
+    :param path: Path to the repository
+    :param logger: Instance of Logger
+    :return:
+    """
+    try:
+        common_instance = Common(logger=logger)
+        common_instance.run(
+            command=Command('git', 'checkout', ref), cwd=path)
+    except RunError as err:
+        logger.print("Failed to do checkout in the repository!", color="red")
+        raise AttributeError from err
+
+
+def clone_repository(url: str, logger: Logger, ref: str) -> None:
```

Comment:
I wonder if `ref` couldn't be optional so we could clone repos with default

Reply:
@tkoscieln I've tried this: https://github.com/teemtee/web/pull/4/commits/0c6afd9d4152f94c95331e9409fd2ad0baa1a6e6, though not sure if I didn't miss anything.

Reply:
Looks good to me.